flight/src/types.rs

```diff
@@ -0,0 +1,39 @@
+use cid::Cid;
+use serde::{Deserialize, Serialize};
+use serde_bytes::ByteBuf;
+
+#[derive(Debug, Clone, Serialize, Deserialize)]
+pub enum ConclusionEvent {
+    Data(ConclusionData),
+    Time(TimeEvent),
+}
+
+#[derive(Debug, Clone, Serialize, Deserialize)]
+pub struct ConclusionInit {
+    pub stream_type: String,
+    pub controllers: String,
```

Comment:
Should rename this to `controller` since we are stating there is only ever going to be one?